library/x509_crl.c

```diff
@@ -418,12 +442,37 @@ int mbedtls_x509_crl_parse_der( mbedtls_x509_crl *chain,
      *                                   -- if present, MUST be v2
      *                        } OPTIONAL
      */
-    if( ( ret = x509_get_entries( &p, end, &crl->entry ) ) != 0 )
+    entry = mbedtls_calloc( 1, sizeof( mbedtls_x509_crl_entry ) );
+    if( entry == NULL )
+    {
+        mbedtls_x509_crl_free( crl );
+        return( MBEDTLS_ERR_X509_ALLOC_FAILED );
+    }
+    count = 1;
```

Comment:
I moved the entire logic into x509_get_entries(). The latter now receives the `crl` object as argument and can therefore hide the entire linked_list-to-array mechanism.